src/library_html5_webgl.js

```diff
@@ -363,6 +363,14 @@ var LibraryHtml5WebGL = {
 #endif
 
     var ext = context.GLctx.getExtension(extString);
+    if (extString == 'WEBGL_multi_draw') {
+      // WEBGL_multi_draw is targeted at both WebGL 1 and WebGL 2.
+      // We now didn't list WEBGL_multi_draw as the automatically enabled extension now.
+      // As there's only one browser implementation so we don't know if enabling it would
+      // have extra overhead yet. This extension exposes extra functions so we need to attach
+      // them to multiDrawWebgl under GLctx object for future reference when calling.
```

Comment:
is this aspect unique to this extension? If not then I'm surprised we didn't need something similar for other ones, and so there may be a better way than this. I can't think of one though. Anyhow, if @juj has a better idea later we can improve this then.

Reply:
> Is this aspect unique to this extension?
Yes. `WEBGL_multi_draw` and `WEBGL_*_base_vertex_base_instance` are still in draft. We add the implementation to hopefully get more clients feedback like unity and skia to help push forward these extensions to community approval.

Enabling this extension automatically when available (like most of the extensions) has no overhead in chromium. And that's for sure. We don't know for Firefox just because there's no implementation yet.

In https://github.com/emscripten-core/emscripten/pull/11054 @juj refactored some extensions that are only available to WebGL 1 context (or only available to WebGL 2 context) to have separate explicit enabling logic to prevent redundant code size. (say the user only targets at WebGL 2 context, having the code auto enabling extensions only available to WebGL 1 makes no sense)
But `WEBGL_multi_draw` is available to both WebGL 1 and 2 context so I don't think this falls into that category.

Maybe I'm making things overcomplicated here...

Reply:
To follow the same pattern as the other extensions above (ANGLE_instanced_arrays, OES_vertex_array_object, etc.), this should not be handled here like this, as it would increase code size for code that does not use multi_draw.

Instead, let's create a new function `_webgl_enable_WEBGL_multi_draw`, and have

```js
    if (extString == 'WEBGL_multi_draw') __webgl_enable_WEBGL_multi_draw(GLctx);
```

and then also have a dedicated function
```js
  emscripten_webgl_enable_WEBGL_multi_draw__deps: ['_webgl_enable_WEBGL_multi_draw'],
  emscripten_webgl_enable_WEBGL_multi_draw: function(ctx) {
    return __webgl_enable_WEBGL_multi_draw(GL.contexts[ctx].GLctx);
  },
```
and in html5_webgl.h,
```h
extern EM_BOOL emscripten_webgl_enable_WEBGL_multi_draw(EMSCRIPTEN_WEBGL_CONTEXT_HANDLE context);
```

That way introducing the extension will not promote code size creep.

Also, even though enabling these extensions might cause slowdowns even when not used, we should add these to be automatically enabled in `GL.initExtensions()` function. That is, specifically, also call `__webgl_enable_WEBGL_multi_draw()` from within `GL.initExtensions()`. This is because that is the exact reason that `GL_SUPPORT_AUTOMATIC_ENABLE_EXTENSIONS` exists: it should enable all extensions to emulate desktop GLES behavior. (codebases that worry about such performance degradation should be building without that flag enabled)

Reply:
> In #11054 @juj refactored some extensions that are only available to WebGL 1 context (or only available to WebGL 2 context) to have separate explicit enabling logic to prevent redundant code size. (say the user only targets at WebGL 2 context, having the code auto enabling extensions only available to WebGL 1 makes no sense)
But WEBGL_multi_draw is available to both WebGL 1 and 2 context so I don't think this falls into that category.

Specifically, the logic to avoid extra code size is not just about whether user is doing WebGL 1 only, or WebGL 2 only, but also whether user is doing WebGL2-without-multi-draw: i.e. if user wants to target WebGL 1 and/or 2, but does not care about the multi_draw feature at all, we should not be emitting extra code size to regress their build output sizes.

Reply:
@shrekshao could you please address this feedback next?

Reply:
Thanks for providing this detailed explanation. Sorry I missed these so was a bit confused. But thanks for highlighting them in the latest comment.
Fixed this. Also the one for base_vertex_base_instance seems to be missing. Fixed as well.
This commit is specially fixing this extension enabling issue: https://github.com/emscripten-core/emscripten/pull/11650/commits/27f3f8e12b2cbda50bbffbead01988b3ece1cc7b